backend/src/utils/actions.ts

```diff
@@ -1,68 +1,86 @@
 import * as RPC from "@explorer/common/types/rpc";
 
-export type Action =
-  | {
-      kind: "createAccount";
-      args: {};
-    }
-  | {
-      kind: "deployContract";
-      args: {
-        code: string;
-      };
-    }
-  | {
-      kind: "functionCall";
-      args: {
-        methodName: string;
-        args: string;
-        gas: number;
-        deposit: string;
-      };
-    }
-  | {
-      kind: "transfer";
-      args: {
-        deposit: string;
-      };
-    }
-  | {
-      kind: "stake";
-      args: {
-        stake: string;
-        publicKey: string;
-      };
-    }
-  | {
-      kind: "addKey";
-      args: {
-        publicKey: string;
-        accessKey: {
-          nonce: number;
-          permission:
-            | {
-                type: "fullAccess";
-              }
-            | {
-                type: "functionCall";
-                contractId: string;
-                methodNames: string[];
-              };
-        };
-      };
-    }
-  | {
-      kind: "deleteKey";
-      args: {
-        publicKey: string;
-      };
-    }
-  | {
-      kind: "deleteAccount";
-      args: {
-        beneficiaryId: string;
-      };
+type CreateAccountAction = {
+  kind: "createAccount";
+  args: {};
+};
+type DeployContractAction = {
+  kind: "deployContract";
+  args: {
+    code: string;
+  };
+};
+type FunctionCallAction = {
+  kind: "functionCall";
+  args: {
+    methodName: string;
+    args: string;
+    gas: number;
+    deposit: string;
+  };
+};
+type TransferAction = {
+  kind: "transfer";
+  args: {
+    deposit: string;
+  };
+};
+type StakeAction = {
+  kind: "stake";
+  args: {
+    stake: string;
+    publicKey: string;
+  };
+};
+type AddKeyAction = {
+  kind: "addKey";
+  args: {
+    publicKey: string;
+    accessKey: {
+      nonce: number;
+      permission:
+        | {
+            type: "fullAccess";
+          }
+        | {
+            type: "functionCall";
+            contractId: string;
+            methodNames: string[];
+          };
     };
+  };
+};
+type DeleteKeyAction = {
+  kind: "deleteKey";
+  args: {
+    publicKey: string;
+  };
+};
+type DeleteAccountAction = {
+  kind: "deleteAccount";
+  args: {
+    beneficiaryId: string;
+  };
+};
+
+type NonDelegateAction =
+  | CreateAccountAction
+  | DeployContractAction
+  | FunctionCallAction
+  | TransferAction
+  | StakeAction
+  | AddKeyAction
+  | DeleteKeyAction
+  | DeleteAccountAction;
+
+type DelegateAction = {
+  kind: "delegate";
+  args: {
```

Comment:
From what I can gather the list of arguments should be bigger, including:
- sender_id
- receiver_id
- actions +
- nonce
- max_block_height
- public_key

Sorry if you just skipped it during the draft and you was going to implement it later.

Reply:
As we don't know what we are going to render on frontend - I didn't pass any of the props to the mapped action.

Reply:
@khorolets Do you feel like we need to add anything more than we have at the moment?